staging/src/k8s.io/component-base/metrics/prometheusextension/timing_histogram_layered.go

```diff
@@ -0,0 +1,142 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package prometheusextension
+
+import (
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	dto "github.com/prometheus/client_model/go"
+)
+
+// NewTimingHistogram creates a new TimingHistogram
```

Comment:
I think I'm missing something, but I don't understand what TimingHistogramLayered is

Reply:
At the moment, this PR has two implementations of TimingHistogram: a layered one that is strictly a client of a WeightedHistogram, and a direct one that shares the WeightedHistogram implementation's mutex and calls a non-public update method.  This is so that the two can be compared.  Based on that comparison, my current position is that we should go forward with the direct one and delete the layered one.  But I wanted to get some review before doing the delete.

Reply:
OK - got it.

Yeah - I agree it makes sense to just delete this one and go ahead with the direct one.